src/melee.cpp

```diff
@@ -1940,3 +1945,129 @@ double player::unarmed_value() const
     // TODO: Martial arts
     return melee_value( ret_null );
 }
+
+void player::disarm( npc &target )
+{
+    if( !target.is_armed() ) {
+        return;
+    }
+
+    item &it = target.weapon;
+
+    const char *target_name = target.name.c_str();
+    const char *item_name = it.tname().c_str();
```

Comment:
No, that's even worse. This would suffice:
```c++
const std::string target_name = target.name;
const std::string item_name = it.tname();
add_msg( _( "You forcefully take %s from %s!" ), item_name.c_str(), target_name.c_str() );
```
Another option (IMHO preferable) is using them directly: 
```c++
add_msg( _( "You forcefully take %s from %s!" ), it.tname().c_str(), target.name.c_str() );
```
The overhead is neglectable, pros outweigh:
- if something in your function makes them change, you still get the actual names.
- the line can be cut-pasted elsewhere more easily.
- it's simply shorter (minus two lines).